File: go/2023/day05/main.go
package main

import (
	"day05/rangemapchain"
	"log"
	"os"
)

func main() {
	part1()
	part2()
}

func part1() {
	f, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()
	rangeMapChain := rangemapchain.NewRangeMapChain(f, true)
	l := rangeMapChain.LowestNumberLocation()
	println("Lowest number location", l)
}

func part2() {
	f, err := os.Open("input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()
	rangeMapChain := rangemapchain.NewRangeMapChain(f, false)
	l := rangeMapChain.LowestNumberLocation()
	println("Lowest number location with seed ranges", l)
}

File: rust/2023/day05/src/main.rs
use std::fs;

use crate::seedlocmap::seedlocmap::SeedLocMap;

mod seedlocmap;

fn main() {
    let seed_loc_map = SeedLocMap::new(fs::File::open("input.txt").unwrap(), false);
    println!("{:?}", seed_loc_map);
    println!("{}", seed_loc_map.lowest_seed_loc());
}
